### PR TITLE
DP-8362 Update most/all Mayflower icons to use the Twig icon() function

### PR DESCRIPTION
**Pull request by ygannett**
Note:
Need to fix the link arrow display on home page banner in FF.  The style is not applied properly.

## Description

Replace the svg icon templates and the template calls with the icon() function.

Associated Drupal PR:
[DP-8362 Update most/all Mayflower icons to use the Twig icon() function #2128](https://github.com/massgov/mass/pull/2128)

## Related Issue / Ticket

- [DP-8362 Update most/all Mayflower icons to use the Twig icon() function](https://jira.state.ma.us/browse/DP-8362)

## Steps to Test

1. Go to any pages with icons.
2. Compare the page with one in `https://mayflower.digital.mass.gov/` and all icons are displayed as they are supposed to.
3. Check source for the icons for their markup.  You'll see 2 sets of `<svg>`.  The first one contains `<use>` as its child.  The second one has `<symbol>` for its child.
mf-render

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

## Additional Notes:

Anything else to add?

#### Impacted Areas in Application

* 

#### @TODO

*

### CODE REVIEW

File: styleguide/source/_patterns/03-organisms/by-author/action-card.twig [hunk 1]
```diff
@@ -9,15 +9,15 @@
     {% for action in actionCard.actions %}
       <div class="ma__action-card__item js-clickable {{action.theme ? 'ma__action-card__item--' ~ action.theme : ''}}">
         <div class="ma__action-card__icon">
-          {% include action.icon %}
+          {{ icon('action.icon') }}
```
Comment (rbayliss) [inline styleguide/source/_patterns/03-organisms/by-author/action-card.twig]: This is the problem, @ygannett.  There is no icon called `action.icon`.  I think you want `{{ icon(action.icon) }}` - no quotes.

**Comment by rbayliss** [comment]
@clairesunstudio @isaacchansky - Just wanted to give you guys a heads up this is coming down the pipe - it's the followup to the icon work that was done on download links a couple weeks ago.  Most notably, we're removing the Twig templates for the icons here.  This may or may not be a BC break, depending on how we're thinking about it.  We still have BC layers in place so that if you try to pass one of the old names (eg: `@atoms/05-icons/svg-phone.twig`) into the `icon` function, it'll map directly to the SVG file, so this shouldn't cause breakage on the Drupal site.

I'd like to get this merged and shipped as soon as we can.  Are you ok with this going out in a minor release in its current state, or should we restore the legacy icon Twig templates until the next major?

**Comment by isaacchansky** [comment]
I'm good with this going out in a minor release

**Comment by rbayliss** [comment]
This looks great!  I think we're ready for @mrossi113's review here.  Melissa, this is the Mayflower PR, which can be pulled into the Drupal PR (https://github.com/massgov/mass/pull/2128) using `composer require massgov/mayflower-artifacts:dev-DP-8362_svg-opt-r2`.

**Review by rbayliss — APPROVED** [review]

**Comment by rbayliss** [comment]
Actually, I just had a thought, which is that removing the icon Twig templates could cause us some heartache on deployment.  If we had to revert the Drupal PR for any reason, we'd also have to revert the Mayflower update (and any other PRs that depend on the Mayflower update).  I don't think we want to put ourselves in that situation, so I'm going to restore the Twig files, which we can remove in a followup.

**Comment by mrossi113** [comment]
@rbayliss is this good to review now?

**Comment by rbayliss** [comment]
@mrossi113, yeah.  It's failing on an unrelated change that I'm working out now.  Whatever the fix ends up being, it shouldn't impact the functionality here.

**Review by mrossi113 — CHANGES_REQUESTED** [review]
The generic icon seems to be appearing in multiple location within Drupal. I will add comments to each section in this PR.

**Comment by mrossi113** [comment]
How-To the How to apply section
screen shot 2018-04-18 at 10 17 06 am


Topic page banner icon
screen shot 2018-04-18 at 10 24 05 am

Service page social media icons
screen shot 2018-04-18 at 10 26 11 am

Footer
screen shot 2018-04-18 at 10 29 45 am

Parks location page
screen shot 2018-04-18 at 10 26 42 am

**Review by mrossi113 — APPROVED** [review]

**Comment by mrossi113** [comment]
LGTM after the fix for mayflower artifacts!

**Pull request merged**